src/banded/BandedMatrix.jl

```diff
@@ -1007,3 +1007,45 @@ function resize(A::BandedSubBandedMatrix, n::Integer, m::Integer)
     l,u = bandwidths(A)
     _BandedMatrix(reshape(resize!(vec(copy(bandeddata(A))), (l+u+1)*m), l+u+1, m), n, l,u)
 end
+
+function sum(A::BandedMatrix; dims=:)
+    if dims isa Colon
+        l, u = bandwidths(A)
+        ret = zero(eltype(A))
+        if l + u < 0
+            return ret
+        end
+        n, m = size(A)
+        for i = 1:n, j = rowrange(A, i)
+            ret += A[i, j]
+        end
+        ret
+    elseif dims > 2
+        A
```

Comment:
Add a test for this special case

Reply:
I have a test towards the bottom for dims = 3

Reply:
ah good. Yes codecov struggles with returned lines.